packages/pigeon/tool/shared/generation.dart

```diff
@@ -0,0 +1,228 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:path/path.dart' as p;
+
+import 'process_utils.dart';
+
+enum GeneratorLanguages {
+  cpp,
+  java,
+  kotlin,
+  objc,
+  swift,
+}
+
+// A map of pigeons/ files to the languages that they can't yet be generated
+// for due to limitations of that generator.
```

Comment:
This is a behavioral change; we used to generate an opted-in list of files for each language, whereas now we generate everything that doesn't actively fail. That means we're generating files that we aren't using yet, but
- that should be harmless
- it can be useful for manual generator change checking
- it'll be one less thing to change when backfilling missing platform tests
- https://github.com/flutter/flutter/issues/115169 will render the difference moot anyway